lsp/src/Language/LSP/VFS.hs

```diff
@@ -92,6 +92,11 @@ data VirtualFile = VirtualFile
   -- remains in the map.
   , _file_text :: !Rope
   -- ^ The full contents of the document
+  , _language_id :: !(Maybe J.LanguageKind)
+  -- ^ The text document's language identifier
+  -- This is a Maybe, since lsp-test shares the VFS with lsp
+  -- and the server sends WorkspaceEdit notifications without a language kind.
+  -- In practical use, this should always be Just.
```

Comment:
```suggestion
  -- This is a Maybe, since when we use the VFS as a client
  -- we don't have this information, since server sends WorkspaceEdit 
  -- notifications without a language kind.
  -- When using the VFS in a server, this should always be Just.
```